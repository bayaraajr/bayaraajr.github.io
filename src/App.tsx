import React, { FC, lazy, Suspense } from 'react';

/* Dependency imports */
import { BrowserRouter, Route, Routes } from 'react-router-dom';

/* Project imports */

const Home = lazy(() => import(/*webpackChunkName: "home" */ "pages/home"));
const HiFive = lazy(() => import(/*webpackChunkName: "hi-five" */ "pages/hi-five"));

interface AppProps {
    [key: string]: any
}

const App: FC<AppProps> = ({ ...props }) => {
    return <div>
        <Suspense fallback={<p>Loading page...</p>}>
            <BrowserRouter>
                <Routes>
                    <Route path="/" element={<Home />} />
                    <Route path="/hifive" element={<HiFive />} />
                </Routes>
            </BrowserRouter>
        </Suspense>
    </div>
}

export default App;