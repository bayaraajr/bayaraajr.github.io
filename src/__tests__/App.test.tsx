import React from 'react';
import App from '../App';
import { shallow } from 'enzyme';

it('Application should mount successfully', () => {
    const comp = shallow(<App />);
    expect(comp).toBeDefined();
})