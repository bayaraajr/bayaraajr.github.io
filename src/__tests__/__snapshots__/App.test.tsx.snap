// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application should mount successfully 1`] = `
<div>
  <p>
    Loading page...
  </p>
</div>
`;

exports[`Should mount successfully 1`] = `
<div>
  <p>
    Loading page...
  </p>
</div>
`;
